datashader/pipeline.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import absolute_import, division, print_function
+
+"""
+Declarative interface to Datashader.  
+
+Provides a configurable pipeline that makes it simpler to specify
+individual stages independently from the others, Currently does not
+add any new underlying capabilities, with all functionality also
+available using the default imperative interface provided by other
+files.
+"""
+
+import param
+
+from . import transfer_functions as tf
+from . import reductions
+from . import core
+
+
+class Interpolate(param.Parameterized):
+    """
+    Parameterized function object to interpolate colors from a scalar input.
+    """
+
+    low = param.Parameter(default="lightpink", doc="""
+         Color string or tuple specifying the starting point for interpolation.""")
+
+    high = param.Parameter(default="red", doc="""
+         Color string or tuple specifying the ending point for interpolation.""")
+
+    how = param.Parameter(default="log", doc="""
+         Function object or string specifying how to map from a scalar into color space.""")
+
+    def __call__(self, agg):
+            return tf.interpolate(agg, self.low, self.high, self.how)
```

Comment:
Ah, in that case you are correct.  The indentation got all screwed up when pasting from Jupyter, and I didn't catch that one.  Annoying!  I've fixed it now; sounds like we can agree on the whitespace convention.  Yes, there's absolutely no reason for indenting ``__call__`` like that.